src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -4864,7 +4946,7 @@ private boolean giveWarning(Type from, Type to) {
 
     private boolean containsTypeEquivalent(Type t, Type s) {
         return isSameType(t, s) || // shortcut
-            containsType(t, s) && containsType(s, t);
+            containsType(t, s, containsTypeUnchecked) && containsType(s, t, containsTypeUnchecked);
```

Comment:
Is this the right replacement? It seems to me that the answer is "it depends". This is called both by "hasSameArgs" which is used to determine overriding (in which case a loose semantics could be ok), but also by the isSameType visitor - in which case the loose semantics should only be enabled conditionally.

Reply:
yep you are right, fixed